buffers.kak

```diff
@@ -32,6 +32,20 @@ hook global WinDisplay .* %{
   set-option global current_bufname %val{bufname}
 }
 
+# A value of 'alt' is recommended, as this is an alias used by multiple filetype rc scripts.
+# NOTE: This is set by default to a command that's expected to fail in order to retain the original plugin behavior.
+# The guid suffix is trying to "guarantee" that it won't accidentally be a command someone might set.
+declare-option str buffer_alternate_command 'nosuchcommand_d1e3a0d4717a4c699eb0c275bbfccd36'
+define-command buffer-alternate -docstring 'open the alternate or previous file in a new buffer' %{
+    try %{
+        # attempt to execute a custom command specified by the `buffer_alternate_command` option
+        evaluate-commands %sh{printf "%s\n" "$kak_opt_buffer_alternate_command"}
```

Comment:
Thanks for this patch, I have some complaints ;)

- I think making features require configuration is not really appropriate
  for plugin. I like it because I can install it, add a mapping for
  `: enter-user-mode buffers<ret>`, and I'm done.  Plus, this behavior is already
  easy to set up with a quick `map global buffers a ': alt<ret>'`.
- `:alt` toggles between test/implementation or *.c/*.h file. OTOH, `ga`
  switches to the last buffer. I use both of them, so I'd like to bind both.

 I've resorted to typing `:alt` but maybe we can add these mappings?

```kak
map global buffers a 'ga'			-docstring 'alternate buffer ↔'
map global buffers t ': alt<ret>'		-docstring 'alternate file ↔'
```

Reply:
Thanks for the response!

I can appreciate not wanting to add configuration for the plugin, and recognize that I can accomplish what I'm after fairly easily in my local config. So then is there anything left to change in the plugin itself?

Bottom line: I think `ba` should exec `: alt<ret>` instead of `ga` since it's just as easy to use `ga` to accomplish `ga`. That way, `ba` and `ga` each do something useful and different (there are only so many 2-key combos available).

I should mention 3 primary motives for this PR:
1. The included `ba` keybinding provided by the buffers user mode is redundant to the built-in `ga` key binding. And there is another "previous" buffers mapping, `bp`.
2. I think that the term "alternate" is misleading and inaccurate if it really means "previous", or "last".
3. The `:alt` alias exists for multiple file types, and I think it makes sense for the buffers user-mode to take advantage of this.

My opinionated change would be to modify the `ba` mapping to use the `alt` command instead of `ga`. Adding a new mapping instead, like you suggest, would be another valid way to utilize the existence of `:alt` in this plugin, though I'd just as soon change `ba` in my own config.

In either case, there are 2 follow up questions to consider:
- What should happen if the `alt` is not defined?
- What should happen If `alt` is defined, but throws an error (no alternate file found or something)?

If in a context in which no `alt` command exists, then `ga` is a possible fallback that should always work. The reason I added a custom command for `ba` mapping instead of just `: alt<ret>` was to define this behavior. However, I could see someone wanting to just see the echo'd error message instead.

Reply:
I see, that all makes sense! Changing the binding to always run `:alt` sounds promising (I'd make it error if unset).
`ga` should probably still be in there, for discoverability (?). The descriptions "last" and "previous" are already taken, so maybe something like "toggle".. I'm not sure